pkg/volume/util/resize_util.go

```diff
@@ -46,45 +50,154 @@ func ClaimToClaimKey(claim *v1.PersistentVolumeClaim) string {
 	return fmt.Sprintf("%s/%s", claim.Namespace, claim.Name)
 }
 
+// UpdatePVSize updates just pv size after cloudprovider resizing is successful
+func UpdatePVSize(
+	pv *v1.PersistentVolume,
+	newSize resource.Quantity,
+	kubeClient clientset.Interface) error {
+	pvClone := pv.DeepCopy()
+
+	oldData, err := json.Marshal(pvClone)
+	if err != nil {
+		return fmt.Errorf("Unexpected error marshaling old PV %q with error : %v", pvClone.Name, err)
+	}
+
+	pvClone.Spec.Capacity[v1.ResourceStorage] = newSize
+
+	newData, err := json.Marshal(pvClone)
+	if err != nil {
+		return fmt.Errorf("Unexpected error marshaling new PV %q with error : %v", pvClone.Name, err)
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, pvClone)
```

Comment:
is resource version in this patch?

Reply:
The PV update is not doing resource version check. I think it is fine, because resource version check is important when you don't want to unconditionally patch a field but is less important when your patch does not depend on more recent version of object. 

In this case - getting latest version of PV object isn't going to make a difference. Also since PV is non-namespaced object, users should not be editing them.